### DISCUSSION TITLE
Bad namings in the useDisclosure hook

### DISCUSSION THREAD

**Discussion by 11janci** [chakra-ui/chakra-ui#6663] [General]
Hi there,

my team was recently implementing a Modal along with a useDisclosure hook. 

The hook returns an object with the following fields (from the docs):
- isOpen
- onClose - Callback function to set a falsy value for the isOpen parameter.
- onOpen - Callback function to set a truthy value for the isOpen parameter.

This is fine to be used for the Modal itself:
```js
<Modal isOpen={isOpen} onClose={onClose}>...</>
```
When for example the user clicks on the close button (`X`), the onClose callback gets called so that the state can be updated accordingly.

However, we also have an API call, at the end of which we want to close the Modal programmatically. Here, we call the onClose() callback:
```js
function onApiFinished() {
  // this actually closes the dialog
  modal.onClose();
}
```
This is a very confusing naming. It suggests that the modal has already been closed, or that some event related to that modal has happened, which is not the case. Having multiple such usages across the code (we have 3 different modals in one component), I never know whether I'm dealing with an actual handler to an event or an action that's gonna trigger that event.

Instead, the hook should separate the callback from setting the `isOpen` state and provide a `close()` and `open()` method. The default callback implementation then should looks sth like
```js
function onClose() {
  if(isOpen) {
    setIsOpen(false);
  }
}
```
I would be curious if you have any comments on this.

Thank you!

**Comment by abdullah-sha256** [comment]
Agreed, it should be called `setOpen` and `setClosed` imo.
